### PR TITLE
fixed warning message in set_channel_types() that caused KeyError whe…

### PR DESCRIPTION
**Pull request by smerdis**
…n unit_old was not a key in the _unit2human dict.

### CODE REVIEW

**Comment by smerdis** [comment]
this is a fix for https://github.com/mne-tools/mne-python/issues/2878

File: mne/channels/channels.py [hunk 1]
```diff
@@ -287,8 +287,10 @@ def set_channel_types(self, mapping):
             unit_old = self.info['chs'][c_ind]['unit']
             unit_new = _human2unit[ch_type]
             if unit_old != _human2unit[ch_type]:
+                if unit_old in _human2unit.keys(): uo_str = _unit2human[unit_old]
+                else: uo_str = unit_old
```
Comment (jasmainak) [inline mne/channels/channels.py]: please follow pep8 conventions. If the key is missing, please add it

Comment (Eric89GXL) [inline mne/channels/channels.py]: FYI you can use dict.get to avoid this sort of conditional in the future

**Comment by smerdis** [comment]
Style should be fixed, and I added the unknown unit to the dict as you suggested. However I'm not sure about this behavior: in the example I presented in #2878, the unit type is 0, and this code would result in a unit type of '0' (with value '0') being added to the dictionary. As another commenter on the issue mentioned, the underlying file reader might be the cause of the weird unit. I had simply intended to make the warning message not cause an error...

**Comment by jasmainak** [comment]
what file reader did you use?

**Comment by jasmainak** [comment]
look up FIFF constants

**Comment by Eric89GXL** [comment]
We actually don't have a `_UNIT_` with value 0, so whatever converter wrote the FIF file wrote some unit type we don't understand. You should be able to add `0: 'unknown'` to the dict around line 182 to fix the problem. Can you try it?

**Pull request closed without merging**